pkg/ccl/testccl/workload/schemachange/schemachange_test: TestWorkload failed [unable to encode table key: *tree.DLTree]

pkg/ccl/testccl/workload/schemachange/schemachange_test.TestWorkload [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/20320486?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/20320486?buildTab=artifacts#/) on master @ [eae530e916a9a77881209f5498870d2831a7bc50](https://github.com/cockroachdb/cockroach/commits/eae530e916a9a77881209f5498870d2831a7bc50):


```
        	            	  | text/template.(*state).evalPipeline
        	            	  | 	GOROOT/src/text/template/exec.go:532
        	            	  | text/template.(*state).walk
        	            	  | 	GOROOT/src/text/template/exec.go:267
        	            	  | text/template.(*state).walk
        	            	  | 	GOROOT/src/text/template/exec.go:280
        	            	  | text/template.(*state).walkIfOrWith
        	            	  | 	GOROOT/src/text/template/exec.go:308
        	            	  | text/template.(*state).walk
        	            	  | 	GOROOT/src/text/template/exec.go:291
        	            	  | text/template.(*state).walk
        	            	  | 	GOROOT/src/text/template/exec.go:280
        	            	  | text/template.(*Template).execute
        	            	  | 	GOROOT/src/text/template/exec.go:224
        	            	  | text/template.(*Template).Execute
        	            	  | 	GOROOT/src/text/template/exec.go:207
        	            	  | text/template.(*Template).ExecuteTemplate
        	            	  | 	GOROOT/src/text/template/exec.go:193
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.Generate[...]
        	            	  | 	pkg/workload/schemachange/generate.go:198
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*operationGenerator).alterTableAlterPrimaryKey
        	            	  | 	pkg/workload/schemachange/operation_generator.go:2825
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*operationGenerator).randOp
        	            	  | 	pkg/workload/schemachange/operation_generator.go:207
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*schemaChangeWorker).runInTxn
        	            	  | 	pkg/workload/schemachange/schemachange.go:501
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*schemaChangeWorker).run
        	            	  | 	pkg/workload/schemachange/schemachange.go:611
        	            	  | pkg/ccl/testccl/workload/schemachange/schemachange_test_test.TestWorkload.TestWorkload.func5.func7
        	            	  | 	pkg/ccl/testccl/workload/schemachange/schema_change_external_test.go:154
        	            	  | golang.org/x/sync/errgroup.(*Group).Go.func1
        	            	  | 	external/org_golang_x_sync/errgroup/errgroup.go:78
        	            	  | runtime.goexit
        	            	  | 	src/runtime/asm_arm64.s:1223
        	            	Wraps: (11) Collect: CollectRows: Query: SELECT * FROM (VALUES ((SELECT true)), ((SELECT true)), ((SELECT NOT EXISTS(SELECT 1 FROM public.table_w0_48 GROUP BY col48_w0_49 HAVING count(*) > 1))), ((SELECT NOT EXISTS(SELECT 1 FROM public.table_w0_48 GROUP BY col48_w0_50 HAVING count(*) > 1))), ((SELECT NOT EXISTS(SELECT 1 FROM public.table_w0_48 GROUP BY col48_w0_51 HAVING count(*) > 1))), ((SELECT NOT EXISTS(SELECT 1 FROM public.table_w0_48 GROUP BY col48_w0_52 HAVING count(*) > 1))), ((SELECT NOT EXISTS(SELECT 1 FROM public.table_w0_48 GROUP BY col48_w0_53 HAVING count(*) > 1))), ((SELECT NOT EXISTS(SELECT 1 FROM public.table_w0_48 GROUP BY col48_w0_55 HAVING count(*) > 1))), ((SELECT NOT EXISTS(SELECT 1 FROM public.table_w0_48 GROUP BY col48_w0_56 HAVING count(*) > 1))), ((SELECT NOT EXISTS(SELECT 1 FROM public.table_w0_48 GROUP BY col48_w0_57 HAVING count(*) > 1))), ((SELECT NOT EXISTS(SELECT 1 FROM public.table_w0_48 GROUP BY col48_w0_58 HAVING count(*) > 1))), ((SELECT NOT EXISTS(SELECT 1 FROM public.table_w0_48 GROUP BY col48_w0_59 HAVING count(*) > 1))), ((SELECT NOT EXISTS(SELECT 1 FROM public.table_w0_48 GROUP BY col48_w0_61 HAVING count(*) > 1))), ((SELECT NOT EXISTS(SELECT 1 FROM public.table_w0_48 GROUP BY col48_w0_62 HAVING count(*) > 1))), ((SELECT NOT EXISTS(SELECT 1 FROM public.table_w0_48 GROUP BY col48_w0_63 HAVING count(*) > 1))), ((SELECT NOT EXISTS(SELECT 1 FROM public.table_w0_48 GROUP BY col48_w0_64 HAVING count(*) > 1))), ((SELECT NOT EXISTS(SELECT 1 FROM public.table_w0_48 GROUP BY col48_w0_65 HAVING count(*) > 1))), ((SELECT NOT EXISTS(SELECT 1 FROM public.table_w0_48 GROUP BY col48_w0_66 HAVING count(*) > 1))), ((SELECT NOT EXISTS(SELECT 1 FROM public.table_w0_48 GROUP BY col48_w0_67 HAVING count(*) > 1)))) as t (res) Args: []
        	            	Wraps: (12) ERROR: internal error: unable to encode table key: *tree.DLTree (SQLSTATE XX000)
        	            	Error types: (1) *markers.withMark (2) *schemachange.ErrorState (3) *withstack.withStack (4) *errutil.withPrefix (5) *withstack.withStack (6) *errutil.withPrefix (7) *withstack.withStack (8) template.ExecError (9) *fmt.wrapError (10) *withstack.withStack (11) *errutil.withPrefix (12) *pgconn.PgError
        	Test:       	TestWorkload
Schema Workload Stats
Total Schema Statements Executed = 134
Total Schema Statements Succeeded = 134
Total Schema Statement Expected Failures = 0
Total Transactions Committed = 87
Total Transactions Rolled Back = 75
Total Transactions Executed = 162
    schema_change_external_test.go:118: backup, tracing data, and system table dumps in /artifacts/tmp/_tmp/67537770b84c3a4ea177539c1e46bd43/logTestWorkload2443774843
    schema_change_external_test.go:127: running DROP with use_declarative_schema_changer = off
    panic.go:636: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/67537770b84c3a4ea177539c1e46bd43/logTestWorkload2443774843
--- FAIL: TestWorkload (167.13s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #151451 pkg/ccl/testccl/workload/schemachange/schemachange_test: TestWorkload failed [C-test-failure O-robot T-sql-foundations branch-release-25.2 release-blocker]
- #151441 pkg/ccl/testccl/workload/schemachange/schemachange_test: TestWorkload failed [C-test-failure O-robot T-sql-foundations branch-release-25.3 release-blocker]
- #150726 pkg/ccl/testccl/workload/schemachange/schemachange_test: TestWorkload failed [box2d must be set or be NULL] [C-test-failure O-robot P-2 T-sql-foundations branch-release-25.2.3-rc]
</p>
</details>
/cc @cockroachdb/sql-foundations
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestWorkload.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

pkg/ccl/testccl/workload/schemachange/schemachange_test.TestWorkload [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/20321640?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/20321640?buildTab=artifacts#/) on master @ [cb2fe9218a4ee8869ec112617429898be3225ba8](https://github.com/cockroachdb/cockroach/commits/cb2fe9218a4ee8869ec112617429898be3225ba8):


```
        	            	  | text/template.(*state).evalPipeline
        	            	  | 	GOROOT/src/text/template/exec.go:532
        	            	  | text/template.(*state).walk
        	            	  | 	GOROOT/src/text/template/exec.go:267
        	            	  | text/template.(*state).walk
        	            	  | 	GOROOT/src/text/template/exec.go:280
        	            	  | text/template.(*state).walkIfOrWith
        	            	  | 	GOROOT/src/text/template/exec.go:308
        	            	  | text/template.(*state).walk
        	            	  | 	GOROOT/src/text/template/exec.go:291
        	            	  | text/template.(*state).walk
        	            	  | 	GOROOT/src/text/template/exec.go:280
        	            	  | text/template.(*Template).execute
        	            	  | 	GOROOT/src/text/template/exec.go:224
        	            	  | text/template.(*Template).Execute
        	            	  | 	GOROOT/src/text/template/exec.go:207
        	            	  | text/template.(*Template).ExecuteTemplate
        	            	  | 	GOROOT/src/text/template/exec.go:193
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.Generate[...]
        	            	  | 	pkg/workload/schemachange/generate.go:198
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*operationGenerator).alterTableAlterPrimaryKey
        	            	  | 	pkg/workload/schemachange/operation_generator.go:2825
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*operationGenerator).randOp
        	            	  | 	pkg/workload/schemachange/operation_generator.go:207
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*schemaChangeWorker).runInTxn
        	            	  | 	pkg/workload/schemachange/schemachange.go:501
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*schemaChangeWorker).run
        	            	  | 	pkg/workload/schemachange/schemachange.go:611
        	            	  | pkg/ccl/testccl/workload/schemachange/schemachange_test_test.TestWorkload.TestWorkload.func5.func7
        	            	  | 	pkg/ccl/testccl/workload/schemachange/schema_change_external_test.go:154
        	            	  | golang.org/x/sync/errgroup.(*Group).Go.func1
        	            	  | 	external/org_golang_x_sync/errgroup/errgroup.go:78
        	            	  | runtime.goexit
        	            	  | 	src/runtime/asm_arm64.s:1223
        	            	Wraps: (11) Collect: CollectRows: Query: SELECT * FROM (VALUES ((SELECT true)), ((SELECT true)), ((SELECT NOT EXISTS(SELECT 1 FROM public.table_w0_279 GROUP BY col279_w0_280 HAVING count(*) > 1))), ((SELECT NOT EXISTS(SELECT 1 FROM public.table_w0_279 GROUP BY col279_w0_281 HAVING count(*) > 1))), ((SELECT NOT EXISTS(SELECT 1 FROM public.table_w0_279 GROUP BY col279_w0_282 HAVING count(*) > 1))), ((SELECT NOT EXISTS(SELECT 1 FROM public.table_w0_279 GROUP BY col279_w0_283 HAVING count(*) > 1))), ((SELECT NOT EXISTS(SELECT 1 FROM public.table_w0_279 GROUP BY col279_w0_284 HAVING count(*) > 1))), ((SELECT NOT EXISTS(SELECT 1 FROM public.table_w0_279 GROUP BY col279_w0_285 HAVING count(*) > 1))), ((SELECT NOT EXISTS(SELECT 1 FROM public.table_w0_279 GROUP BY col279_w0_286 HAVING count(*) > 1))), ((SELECT NOT EXISTS(SELECT 1 FROM public.table_w0_279 GROUP BY col279_w0_287 HAVING count(*) > 1))), ((SELECT NOT EXISTS(SELECT 1 FROM public.table_w0_279 GROUP BY col279_w0_288 HAVING count(*) > 1))), ((SELECT NOT EXISTS(SELECT 1 FROM public.table_w0_279 GROUP BY col279_w0_290 HAVING count(*) > 1))), ((SELECT NOT EXISTS(SELECT 1 FROM public.table_w0_279 GROUP BY col279_w0_291 HAVING count(*) > 1))), ((SELECT NOT EXISTS(SELECT 1 FROM public.table_w0_279 GROUP BY col279_w0_293 HAVING count(*) > 1))), ((SELECT NOT EXISTS(SELECT 1 FROM public.table_w0_279 GROUP BY col279_w0_294 HAVING count(*) > 1))), ((SELECT NOT EXISTS(SELECT 1 FROM public.table_w0_279 GROUP BY col279_w0_295 HAVING count(*) > 1))), ((SELECT NOT EXISTS(SELECT 1 FROM public.table_w0_279 GROUP BY col279_w0_296 HAVING count(*) > 1))), ((SELECT NOT EXISTS(SELECT 1 FROM public.table_w0_279 GROUP BY col279_w0_297 HAVING count(*) > 1)))) as t (res) Args: []
        	            	Wraps: (12) ERROR: internal error: unable to encode table key: *tree.DLTree (SQLSTATE XX000)
        	            	Error types: (1) *markers.withMark (2) *schemachange.ErrorState (3) *withstack.withStack (4) *errutil.withPrefix (5) *withstack.withStack (6) *errutil.withPrefix (7) *withstack.withStack (8) template.ExecError (9) *fmt.wrapError (10) *withstack.withStack (11) *errutil.withPrefix (12) *pgconn.PgError
        	Test:       	TestWorkload
Schema Workload Stats
Total Schema Statements Executed = 136
Total Schema Statements Succeeded = 136
Total Schema Statement Expected Failures = 0
Total Transactions Committed = 85
Total Transactions Rolled Back = 76
Total Transactions Executed = 161
    schema_change_external_test.go:118: backup, tracing data, and system table dumps in /artifacts/tmp/_tmp/67537770b84c3a4ea177539c1e46bd43/logTestWorkload432283585
    schema_change_external_test.go:127: running DROP with use_declarative_schema_changer = on
    panic.go:636: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/67537770b84c3a4ea177539c1e46bd43/logTestWorkload432283585
--- FAIL: TestWorkload (108.92s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #151451 pkg/ccl/testccl/workload/schemachange/schemachange_test: TestWorkload failed [C-test-failure O-robot T-sql-foundations branch-release-25.2 release-blocker]
- #151441 pkg/ccl/testccl/workload/schemachange/schemachange_test: TestWorkload failed [C-test-failure O-robot T-sql-foundations branch-release-25.3 release-blocker]
- #150726 pkg/ccl/testccl/workload/schemachange/schemachange_test: TestWorkload failed [box2d must be set or be NULL] [C-test-failure O-robot P-2 T-sql-foundations branch-release-25.2.3-rc]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestWorkload.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

pkg/ccl/testccl/workload/schemachange/schemachange_test.TestWorkload [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsIbmCloudLinuxS390x_UnitTestsS390x/20322957?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsIbmCloudLinuxS390x_UnitTestsS390x/20322957?buildTab=artifacts#/) on master @ [34acbb7ce2bfb0f508a03f5ac333818883b8c8e7](https://github.com/cockroachdb/cockroach/commits/34acbb7ce2bfb0f508a03f5ac333818883b8c8e7):


```
        	            	  | text/template.(*state).evalPipeline
        	            	  | 	GOROOT/src/text/template/exec.go:532
        	            	  | text/template.(*state).walk
        	            	  | 	GOROOT/src/text/template/exec.go:267
        	            	  | text/template.(*state).walk
        	            	  | 	GOROOT/src/text/template/exec.go:280
        	            	  | text/template.(*state).walkIfOrWith
        	            	  | 	GOROOT/src/text/template/exec.go:308
        	            	  | text/template.(*state).walk
        	            	  | 	GOROOT/src/text/template/exec.go:291
        	            	  | text/template.(*state).walk
        	            	  | 	GOROOT/src/text/template/exec.go:280
        	            	  | text/template.(*Template).execute
        	            	  | 	GOROOT/src/text/template/exec.go:224
        	            	  | text/template.(*Template).Execute
        	            	  | 	GOROOT/src/text/template/exec.go:207
        	            	  | text/template.(*Template).ExecuteTemplate
        	            	  | 	GOROOT/src/text/template/exec.go:193
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.Generate[...]
        	            	  | 	pkg/workload/schemachange/generate.go:198
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*operationGenerator).alterTableAlterPrimaryKey
        	            	  | 	pkg/workload/schemachange/operation_generator.go:2825
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*operationGenerator).randOp
        	            	  | 	pkg/workload/schemachange/operation_generator.go:207
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*schemaChangeWorker).runInTxn
        	            	  | 	pkg/workload/schemachange/schemachange.go:501
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*schemaChangeWorker).run
        	            	  | 	pkg/workload/schemachange/schemachange.go:611
        	            	  | pkg/ccl/testccl/workload/schemachange/schemachange_test_test.TestWorkload.TestWorkload.func5.func7
        	            	  | 	pkg/ccl/testccl/workload/schemachange/schema_change_external_test.go:154
        	            	  | golang.org/x/sync/errgroup.(*Group).Go.func1
        	            	  | 	external/org_golang_x_sync/errgroup/errgroup.go:78
        	            	  | runtime.goexit
        	            	  | 	src/runtime/asm_s390x.s:797
        	            	Wraps: (11) Collect: CollectRows: Query: SELECT * FROM (VALUES ((SELECT NOT EXISTS(SELECT 1 FROM public.table_w1_91 GROUP BY col91_w1_92 HAVING count(*) > 1))), ((SELECT NOT EXISTS(SELECT 1 FROM public.table_w1_91 GROUP BY col91_w1_93 HAVING count(*) > 1))), ((SELECT NOT EXISTS(SELECT 1 FROM public.table_w1_91 GROUP BY col91_w1_94 HAVING count(*) > 1))), ((SELECT NOT EXISTS(SELECT 1 FROM public.table_w1_91 GROUP BY col91_w1_95 HAVING count(*) > 1))), ((SELECT NOT EXISTS(SELECT 1 FROM public.table_w1_91 GROUP BY col91_w1_96 HAVING count(*) > 1))), ((SELECT NOT EXISTS(SELECT 1 FROM public.table_w1_91 GROUP BY col91_w1_98 HAVING count(*) > 1)))) as t (res) Args: []
        	            	Wraps: (12) ERROR: internal error: unable to encode table key: *tree.DLTree (SQLSTATE XX000)
        	            	Error types: (1) *markers.withMark (2) *schemachange.ErrorState (3) *withstack.withStack (4) *errutil.withPrefix (5) *withstack.withStack (6) *errutil.withPrefix (7) *withstack.withStack (8) template.ExecError (9) *fmt.wrapError (10) *withstack.withStack (11) *errutil.withPrefix (12) *pgconn.PgError
        	Test:       	TestWorkload
Schema Workload Stats
Total Schema Statements Executed = 109
Total Schema Statements Succeeded = 109
Total Schema Statement Expected Failures = 0
Total Transactions Committed = 65
Total Transactions Rolled Back = 56
Total Transactions Executed = 121
    schema_change_external_test.go:118: backup, tracing data, and system table dumps in /artifacts/tmp/_tmp/e1bc47aa61a233e9c8ddfa2c4c769198/logTestWorkload1711372626
    schema_change_external_test.go:127: running DROP with use_declarative_schema_changer = off
    panic.go:636: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/e1bc47aa61a233e9c8ddfa2c4c769198/logTestWorkload1711372626
--- FAIL: TestWorkload (94.82s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #151451 pkg/ccl/testccl/workload/schemachange/schemachange_test: TestWorkload failed [C-test-failure O-robot T-sql-foundations branch-release-25.2 release-blocker]
- #151441 pkg/ccl/testccl/workload/schemachange/schemachange_test: TestWorkload failed [C-test-failure O-robot T-sql-foundations branch-release-25.3 release-blocker]
- #150726 pkg/ccl/testccl/workload/schemachange/schemachange_test: TestWorkload failed [box2d must be set or be NULL] [C-test-failure O-robot P-2 T-sql-foundations branch-release-25.2.3-rc]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestWorkload.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

pkg/ccl/testccl/workload/schemachange/schemachange_test.TestWorkload [failed](https://mesolite.cluster.engflow.com/invocations/default/35e6191f-17ba-4b04-87cc-11f8a789fbcc?testReportRun=3&testReportShard=1&testReportAttempt=1#targets-Ly9wa2cvY2NsL3Rlc3RjY2wvd29ya2xvYWQvc2NoZW1hY2hhbmdlOnNjaGVtYWNoYW5nZV90ZXN0) on master @ [34acbb7ce2bfb0f508a03f5ac333818883b8c8e7](https://github.com/cockroachdb/cockroach/commits/34acbb7ce2bfb0f508a03f5ac333818883b8c8e7):


```
        	            	  | text/template.(*state).evalPipeline
        	            	  | 	GOROOT/src/text/template/exec.go:532
        	            	  | text/template.(*state).walk
        	            	  | 	GOROOT/src/text/template/exec.go:267
        	            	  | text/template.(*state).walk
        	            	  | 	GOROOT/src/text/template/exec.go:280
        	            	  | text/template.(*state).walkIfOrWith
        	            	  | 	GOROOT/src/text/template/exec.go:308
        	            	  | text/template.(*state).walk
        	            	  | 	GOROOT/src/text/template/exec.go:291
        	            	  | text/template.(*state).walk
        	            	  | 	GOROOT/src/text/template/exec.go:280
        	            	  | text/template.(*Template).execute
        	            	  | 	GOROOT/src/text/template/exec.go:224
        	            	  | text/template.(*Template).Execute
        	            	  | 	GOROOT/src/text/template/exec.go:207
        	            	  | text/template.(*Template).ExecuteTemplate
        	            	  | 	GOROOT/src/text/template/exec.go:193
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.Generate[...]
        	            	  | 	pkg/workload/schemachange/generate.go:198
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*operationGenerator).alterTableAlterPrimaryKey
        	            	  | 	pkg/workload/schemachange/operation_generator.go:2825
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*operationGenerator).randOp
        	            	  | 	pkg/workload/schemachange/operation_generator.go:207
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*schemaChangeWorker).runInTxn
        	            	  | 	pkg/workload/schemachange/schemachange.go:501
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*schemaChangeWorker).run
        	            	  | 	pkg/workload/schemachange/schemachange.go:611
        	            	  | pkg/ccl/testccl/workload/schemachange/schemachange_test_test.TestWorkload.TestWorkload.func5.func7
        	            	  | 	pkg/ccl/testccl/workload/schemachange/schema_change_external_test.go:154
        	            	  | golang.org/x/sync/errgroup.(*Group).Go.func1
        	            	  | 	external/org_golang_x_sync/errgroup/errgroup.go:78
        	            	  | runtime.goexit
        	            	  | 	src/runtime/asm_amd64.s:1700
        	            	Wraps: (11) Collect: CollectRows: Query: SELECT * FROM (VALUES ((SELECT true)), ((SELECT true)), ((SELECT true)), ((SELECT true)), ((SELECT true)), ((SELECT true)), ((SELECT true)), ((SELECT NOT EXISTS(SELECT 1 FROM public.table_w0_363 GROUP BY col363_w0_385 HAVING count(*) > 1))), ((SELECT NOT EXISTS(SELECT 1 FROM public.table_w0_363 GROUP BY col363_w0_387 HAVING count(*) > 1))), ((SELECT NOT EXISTS(SELECT 1 FROM public.table_w0_363 GROUP BY col363_w0_388 HAVING count(*) > 1))), ((SELECT NOT EXISTS(SELECT 1 FROM public.table_w0_363 GROUP BY col363_w0_391 HAVING count(*) > 1))), ((SELECT NOT EXISTS(SELECT 1 FROM public.table_w0_363 GROUP BY col363_w0_392 HAVING count(*) > 1))), ((SELECT NOT EXISTS(SELECT 1 FROM public.table_w0_363 GROUP BY col363_w0_393 HAVING count(*) > 1))), ((SELECT NOT EXISTS(SELECT 1 FROM public.table_w0_363 GROUP BY col363_w0_394 HAVING count(*) > 1))), ((SELECT NOT EXISTS(SELECT 1 FROM public.table_w0_363 GROUP BY col363_w0_395 HAVING count(*) > 1))), ((SELECT NOT EXISTS(SELECT 1 FROM public.table_w0_363 GROUP BY col363_w0_397 HAVING count(*) > 1))), ((SELECT NOT EXISTS(SELECT 1 FROM public.table_w0_363 GROUP BY col363_w0_399 HAVING count(*) > 1))), ((SELECT NOT EXISTS(SELECT 1 FROM public.table_w0_363 GROUP BY col363_w0_401 HAVING count(*) > 1))), ((SELECT NOT EXISTS(SELECT 1 FROM public.table_w0_363 GROUP BY col363_w0_403 HAVING count(*) > 1)))) as t (res) Args: []
        	            	Wraps: (12) ERROR: internal error: unable to encode table key: *tree.DLTree (SQLSTATE XX000)
        	            	Error types: (1) *markers.withMark (2) *schemachange.ErrorState (3) *withstack.withStack (4) *errutil.withPrefix (5) *withstack.withStack (6) *errutil.withPrefix (7) *withstack.withStack (8) template.ExecError (9) *fmt.wrapError (10) *withstack.withStack (11) *errutil.withPrefix (12) *pgconn.PgError
        	Test:       	TestWorkload
Schema Workload Stats
Total Schema Statements Executed = 175
Total Schema Statements Succeeded = 175
Total Schema Statement Expected Failures = 0
Total Transactions Committed = 105
Total Transactions Rolled Back = 88
Total Transactions Executed = 193
    schema_change_external_test.go:118: backup, tracing data, and system table dumps in /mnt/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/testlogs/pkg/ccl/testccl/workload/schemachange/schemachange_test/run_3_of_10/test.outputs/logTestWorkload1452330170
    schema_change_external_test.go:127: running DROP with use_declarative_schema_changer = off
    panic.go:636: -- test log scope end --
test logs left over in: outputs.zip/logTestWorkload1452330170
--- FAIL: TestWorkload (112.63s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=3</code>
 - <code>shard=1</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #151451 pkg/ccl/testccl/workload/schemachange/schemachange_test: TestWorkload failed [C-test-failure O-robot T-sql-foundations branch-release-25.2 release-blocker]
- #151441 pkg/ccl/testccl/workload/schemachange/schemachange_test: TestWorkload failed [C-test-failure O-robot T-sql-foundations branch-release-25.3 release-blocker]
- #150726 pkg/ccl/testccl/workload/schemachange/schemachange_test: TestWorkload failed [box2d must be set or be NULL] [C-test-failure O-robot P-2 T-sql-foundations branch-release-25.2.3-rc]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestWorkload.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

--- a/pkg/sql/rowenc/keyside/encode.go
+++ b/pkg/sql/rowenc/keyside/encode.go
@@ -183,6 +183,11 @@ func Encode(b []byte, val tree.Datum, dir encoding.Direction) ([]byte, error) {
 		return encodeJSONKey(b, t, dir)
 	}
 	if buildutil.CrdbTestBuild {
+		if _, isLTree := val.(*tree.DLTree); isLTree {
+			// TODO(paulniziolek): remove this exception once key encoding is
+			// added.
+			return nil, errors.Newf("LTREE key encoding is not implemented yet")
+		}
 		return nil, errors.AssertionFailedf("unable to encode table key: %T", val)
 	}
 	return nil, errors.Errorf("unable to encode table key: %T", val)

